components/content/content-editor/controls/mention/autocomplete/controls.ts

```diff
@@ -0,0 +1,254 @@
+import { Api } from 'game-jolt-frontend-lib/components/api/api.service';
+import { EditorView } from 'prosemirror-view';
+import Vue from 'vue';
+import { Component, Prop, Watch } from 'vue-property-decorator';
+import { numberSort } from '../../../../../../utils/array';
+import AppLoading from '../../../../../../vue/components/loading/loading.vue';
+import { Screen } from '../../../../../screen/screen-service';
+import AppScrollScroller from '../../../../../scroll/scroller/scroller.vue';
+import AppUserAvatarImg from '../../../../../user/user-avatar/img/img.vue';
+import { User } from '../../../../../user/user.model';
+import AppUserVerifiedTick from '../../../../../user/verified-tick/verified-tick.vue';
+import { MentionCache, MentionCacheUser } from '../../../../mention-cache';
+import { ContentEditorService } from '../../../content-editor.service';
+import { ContentEditorSchema } from '../../../schemas/content-editor-schema';
+
+@Component({
+	components: {
+		AppUserAvatarImg,
+		AppUserVerifiedTick,
+		AppScrollScroller,
+		AppLoading,
+	},
+})
+export default class AppContentEditorControlsMentionAutocompleteControls extends Vue {
+	@Prop(Object)
+	view!: EditorView<ContentEditorSchema>;
+	@Prop(Number)
+	stateCounter!: number;
+	@Prop(Number)
+	canShow!: number;
+
+	// We cache prior search results so we don't need to query backend again.
+	private static searchResults: { [k: string]: User[] } = {};
```

Comment:
I don't like static state on the components themselves, even if the suggestions are unbiased global results that are not route/user dependent.

Should we use a store for this?

Also, no harm in being more explicit with the key name: `k` => `search`

Reply:
A service can also work if store is a pain in the butt to do right now.